src/ansitest/plugin.py

```diff
@@ -0,0 +1,508 @@
+# cspell:ignore envlist
+"""tox plugin to emit a github matrix."""
+
+import json
+import logging
+import os
+import re
+import sys
+import uuid
+
+from dataclasses import asdict, dataclass, field
+from pathlib import Path
+from typing import List, Tuple, TypeVar
+
+import yaml
+
+from tox.config.cli.parser import ToxParser
+from tox.config.loader.memory import MemoryLoader
+from tox.config.loader.section import Section
+from tox.config.loader.str_convert import StrConvert
+from tox.config.sets import ConfigSet, CoreConfigSet, EnvConfigSet
+from tox.config.types import EnvList
+from tox.plugin import impl
+from tox.session.state import State
+from tox.tox_env.python.api import PY_FACTORS_RE
+
+
+ALLOWED_EXTERNALS = [
+    "bash",
+    "cp",
+    "git",
+    "rm",
+    "rsync",
+    "mkdir",
+    "cd",
+    "echo",
+]
+ENV_LIST = """
+{integration, sanity, unit}-py3.8-{2.9, 2.10, 2.11, 2.12, 2.13}
+{integration, sanity, unit}-py3.9-{2.10, 2.11, 2.12, 2.13, 2.14, milestone, devel}
+{integration, sanity, unit}-py3.10-{2.12, 2.13, 2.14, milestone, devel}
+{integration, sanity, unit}-py3.11-{2.14, milestone, devel}
```

Comment:
Is it the intention to frequently update this plugin to only support these versions of Ansible? Is the user able to customize the Ansible versions they want to test? (Also, these seem to be `ansible-core` versions and not full Ansible release versions?)

Reply:
Yes, the goal is for this to be maintained.  If we can get some adaption across any of the community collection maintainers or internal ansible content engineering teams, I don't think this should add much operational overhead.

Yes, the goal here now was for collection testing which shouldn't require the full ansible package.  Any collection should have it's deps listed in it's requirements file, which is part of the reason the collection is copied to the tox venv, built, and installed.  the act of installation pulls in those other collection deps.